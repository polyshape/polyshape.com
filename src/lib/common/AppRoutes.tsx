import { Outlet } from "react-router-dom";
import Home from "../../pages/Home";
import Vision from "../../pages/orientation/Vision";
import Journey from "../../pages/orientation/Journey";
import Software from "../../pages/r_n_d/Software";
import Patents from "../../pages/r_n_d/Patents";
import Publications from "../../pages/r_n_d/Publications";
import Dissemination from "../../pages/r_n_d/Dissemination";
import Projects from "../../pages/portfolio/Projects";
import Partners from "../../pages/portfolio/Partners";
import News from "../../pages/News";
import Contact from "../../pages/Contact";
import type { JSX } from "react";

export const AppRoutes = {
  HOME: {
    id: "home",
    path: "/",
    title: "Home",
    element: <Home/>,
    isParent: true,
    children: [] as const
  },
  ORIENTATION: {
    id: "orientation",
    path: "/orientation",
    title: "Orientation",
    element: <Outlet/>,
    isParent: true,
    children: ["VISION", "JOURNEY"] as const
  },
  VISION: {
    id: "vision",
    path: "/orientation/vision",
    title: "Vision",
    element: <Vision/>,
    isParent: false,
    children: [] as const
  },
  JOURNEY: {
    id: "journey",
    path: "/orientation/journey",
    title: "Journey",
    element: <Journey/>,
    isParent: false,
    children: [] as const
  },
  RND: {
    id: "rnd",
    path: "/rnd",
    title: "R&D",
    element: <Outlet/>,
    isParent: true,
    children: ["SOFTWARE", "PATENTS", "PUBLICATIONS", "DISSEMINATION"] as const
  },
  SOFTWARE: {
    id: "software",
    path: "/rnd/software",
    title: "Software",
    element: <Software/>,
    isParent: false,
    children: [] as const
  },
  PATENTS: {
    id: "patents",
    path: "/rnd/patents",
    title: "Patents",
    element: <Patents/>,
    isParent: false,
    children: [] as const
  },
  PUBLICATIONS: {
    id: "publications",
    path: "/rnd/publications",
    title: "Publications",
    element: <Publications/>,
    isParent: false,
    children: [] as const
  },
  DISSEMINATION: {
    id: "dissemination",
    path: "/rnd/dissemination",
    title: "Dissemination",
    element: <Dissemination/>,
    isParent: false,
    children: [] as const
  },
  PORTFOLIO: {
    id: "portfolio",
    path: "/portfolio",
    title: "Portfolio",
    element: <Outlet/>,
    isParent: true,
    children: ["PROJECTS","PARTNERS"] as const
  },
  PROJECTS: {
    id: "projects",
    path: "/portfolio/projects",
    title: "Projects",
    element: <Projects/>,
    isParent: false,
    children: [] as const
  },
  PARTNERS: {
    id: "partners",
    path: "/portfolio/partners",
    title: "Partners",
    element: <Partners/>,
    isParent: false,
    children: [] as const
  },
  NEWS: {
    id: "news",
    path: "/news",
    title: "News",
    element: <News/>,
    isParent: true,
    children: [] as const
  },
  CONTACT: {
    id: "contact",
    path: "/contact",
    title: "Contact",
    element: <Contact/>,
    isParent: true,
    children: [] as const
  }
} as const;

export type AppRoute = {
  id: string;
  path: string;
  title: string;
  element: JSX.Element;
  isParent: boolean;
  children: readonly (keyof typeof AppRoutes)[];
};
