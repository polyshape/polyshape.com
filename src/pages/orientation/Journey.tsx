import { AppRoutes } from "../../lib/common/AppRoutes";

export default function Journey() {
  return (
    <div className="prose">
      <h1 className="hero__title">{AppRoutes.JOURNEY.title}</h1>
      <p>
        PolyShape's journey began in 2023 with research in the UK and in 2025 continued with an additional base in Greece. From the start, the focus has been to build health intelligence on mathematical foundations, to implement these foundations in software, and to translate them into both R&D and consulting for biomedical and healthcare applications.
      </p>
    </div>
  );
}
