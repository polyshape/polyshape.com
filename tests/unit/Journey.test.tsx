import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Journey from '../../src/pages/orientation/Journey';
import { AppRoutes } from '../../src/lib/common/AppRoutes';

describe('Journey page', () => {
  it('renders the main title', () => {
    const { getByText } = render(<Journey />);
    expect(getByText(AppRoutes.JOURNEY.title)).toBeInTheDocument();
  });

  it('applies correct CSS classes', () => {
    const { container } = render(<Journey />);
    expect(container.querySelector('.prose')).toBeInTheDocument();
    expect(container.querySelector('.hero__title')).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { container } = render(<Journey />);
    expect(container).toMatchSnapshot();
  });
});
