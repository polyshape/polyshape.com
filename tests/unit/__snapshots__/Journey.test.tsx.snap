// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Journey page matches the snapshot 1`] = `
<div>
  <div
    class="prose"
  >
    <h1
      class="hero__title"
    >
      Journey
    </h1>
    <p>
      PolyShape's journey began in 2023 with research in the UK and in 2025 continued with an additional base in Greece. From the start, the focus has been to build health intelligence on mathematical foundations, to implement these foundations in software, and to translate them into both R&D and consulting for biomedical and healthcare applications.
    </p>
  </div>
</div>
`;
